src/platforms/esp32/tools/mkimage.sh.in

```diff
@@ -22,11 +22,11 @@
 escript "@CMAKE_BINARY_DIR@/mkimage.erl" \
     --root_dir "@CMAKE_BINARY_DIR@/../../../.." \
     --config "@CMAKE_BINARY_DIR@/mkimage.config" \
-    --out "@CMAKE_BINARY_DIR@/atomvm-@CONFIG_IDF_TARGET@-@ATOMVM_VERSION@.img" \
```

Comment:
I would prefer that we not change the mkimage and flashimage scripts to remove the version.  The reason being that these scripts are also used at dev time, and having the git hashes in the file names (separate from the release) is useful to me, so that I know exactly which version I am about to flash.  I think we can manage this in the release workflow when you rename by using a wildcard or regex to pick out the source file.

Reply:
> The reason being that these scripts are also used at dev time, and having the git hashes in the file names (separate from the release) is useful to me, so that I know exactly which version I am about to flash.

I would like to hear more about your usecase.

Right now I have in my mind 3 different possible workflows:

a. When needing a fast write-and-test cycle: you are working on a NIF/port driver/component/what ever, and you don't want to flash everything. So `idf.py flash monitor` is enough for building, flashing and testing.

b. When needing to be sure to have the lastest (or any specific revision) on your device with all the right matching libraries (flashing time doesn't matter, and erasing everything is ok): `flashimage.sh` is well suited for the task, except one thing:
It is not consistent with the make/idf.py behavior. When I enter `idf.py flash` I expect everything being built before, so I would expect `flashimage.sh` doing `idf.py build` just before making the image and flashing it (without this change a not versioned image might be annoying).

Also I feel like having a versioned image might be misleading when dealing with a dirty git tree.

c. When needing as a user to flash a stable released version: the release page on GitHub is the go-to place.

> I think we can manage this in the release workflow when you rename by using a wildcard or regex to pick out the source file.

Yes, it is possible, but I see a consistency problem: usually build tools just replace previous output, and so does the Pico, the WASM or the unix port. I think we should follow that approach too.

Reply:
So, yes, at dev time, mostly I use case a.  

But if I make a change to the Erlang (or Elixir) code, then I will do an `idf.py build` and a `make` at the top level to align the ESP32 VM and atomvmlib.avm code.  I then do a mkimage, which bundles everything, and a flash.  I guess, technically, having the git hash there is not critical to my dev workflow, upon reflection.  It is just something I am used to.

If it puts the CI at risk of instability, then sure, let's remove the git hash from the mkimage and flashimage.  It does mean that if I wanted to make an image "for posterity" (to save for future reference), I will lose that information, but so far I haven't really had to do that.